### PR TITLE
Simpler conditions for security config check

### PR DESCRIPTION
**Pull request by pornel**
I've received privately a question about logic of this check, and realised it's hard to follow. I've split and reordered it for clarity.

### CODE REVIEW

**Comment by zorgiepoo** [comment]
I actually think the code in `master` is better to reason about. I can look at a particular branch without needing more context in the procedure as opposed to the changes provided here. Returning early in this procedure makes the coder *harder* to understand.

I'm all for adding comments though.. Perhaps the most confusing thing is the is main bundle check and why we care about it.

**Comment by pornel** [comment]
```c
 if (isMainBundle && !(hasPublicDSAKey || hostIsCodeSigned)) { 
 } else if (isMainBundle && !hasPublicDSAKey && !servingOverHttps) {
```

I find reasoning about this very hard. Because it's a cascade of else/if, I need to notice relationship between `!hasPublicDSAKey` and `!(hasPublicDSAKey || hostIsCodeSigned)` and how subtly different it is if only `!hostIsCodeSigned` was in the previous condition.

**Comment by zorgiepoo** [comment]
I'll just note that I mentally translate the first line into `isMainBundle && !hasPublicDSAKey && !hostIsCodeSigned`

The thing is, the code in `master` doesn't *have* to use else if's. It only does that so that less alerts are displayed, but otherwise it's not very significant. I find having this "redundancy" *better* to reason about the code. Bugs from code are prone to happen when order is significant in code, when code has side effects, when portions of code need outside context, when it had to be re-arranged/re-ordered for it to still be correct.

Just my take.

**Comment by pornel** [comment]
How about now?

**Comment by zorgiepoo** [comment]
Yeah, I like it.
